### PR TITLE
change stop() close reader first

### PR DESCRIPTION
**Pull request by wonderflow**
## Changes
   
- [ ] feature1
- [ ] feature2
- [ ] fixbug1
- [ ] fixbug2
 
## Reviewers

- [ ] @[someone] please review
- [ ] @[someotherone] please review

## Wiki Changes
 
- options1...
- options2...
    
## Checklist
   
- [ ] Rebased/mergeable
- [ ] Tests pass
- [ ] Wiki updated

### CODE REVIEW

File: mgr/runner.go [hunk 1]
```diff
@@ -706,26 +706,31 @@ func addTagsToData(tags map[string]interface{}, datas []Data, runnername string)
...
 	}
+
+	log.Infof("Runner[%v] waiting for Run() stopped signal", r.Name())
+	timer := time.NewTimer(time.Second * 10)
+	select {
+	case <-r.exitChan:
+		log.Warnf("runner %v has been stopped", r.Name())
+	case <-timer.C:
+		log.Errorf("runner %v exited timeout, start to force stop", r.Name())
+		atomic.AddInt32(&r.stopped, 1)
```
Comment (Unknwon) [inline mgr/runner.go]: 在方法的第一行 Line 713 有相同的一行，这两行是有不同作用吗？

Comment (wonderflow) [reply mgr/runner.go]: 类似双保险吧

Comment (wonderflow) [reply mgr/runner.go]: 实际上是相同作用

Comment (Unknwon) [reply mgr/runner.go]: 好吧。。加两次是有意义的 https://sourcegraph.com/github.com/qiniu/logkit@5cf9ca92b8e30c626749c44aef1c48b5d12707a7/-/blob/mgr/runner.go#L564

**Comment by Unknwon** [comment]
LGTM

**Pull request merged**